### DISCUSSION TITLE
How to use assert in V repl

### DISCUSSION THREAD

**Discussion by SebastianMpl** [vlang/v#19602] [Questions and Answers] [answered]
Hey Guys,

I'm still having version 0.4 and reading the documentation about assert I found some code that I tried in repl, but it fails to run:

```
Welcome to the V REPL (for help with V itself, type  exit , then run  v help ).
Note: the REPL is highly experimental. For best V experience, use a text editor, 
save your code in a  main.v  file and execute:  v run main.v 
V 0.4.0 ffdd5bb . Use  list  to see the accumulated program so far.
Use Ctrl-C or  exit  to exit, or  help  to see other available commands.
>>> name := 'Bob'
>>> assert name.len == 3
error: invalid expression: unexpected keyword `assert`
    6 | 
    7 | name := 'Bob'
    8 | println(assert name.len == 3)
      |         ~~~~~~
>>> println(assert name.len == 3)
error: invalid expression: unexpected keyword `assert`
    6 | 
    7 | name := 'Bob'
    8 | println(assert name.len == 3)
      |         ~~~~~~
>>> static assert name.len == 3
error: the `static` keyword is invalid here
    6 | 
    7 | name := 'Bob'
    8 | println(static assert name.len == 3)
      |         ~~~~~~
>>> 

```

Does that mean that the assert keyword can't be used in repl, or Am I doing something wrong?

Thank you

**Comment by JalonSolov** [comment, marked as answer]
The REPL is very limited, and does not work as most people think.  You are better off either writing the code in a .v file, or using the `-e` option on the command line.

**Reply by SebastianMpl** [reply]
@JalonSolov 
Thank you for quick reply. The -e option works fine and that's enough for my needs.

**Comment by medvednikov** [comment]
@JalonSolov in this case it has nothing to do with REPL,  this is simply incorrect V:

```
a.v:6:10: error: invalid expression: unexpected keyword `assert`
    4 |
    5 | fn main() {
    6 |     println(assert 3 == 3)
      |             ~~~~~~
    7 | }
    8 |
```

assert is not an expression

**Reply by medvednikov** [reply]
@SebastianMpl use `println(3 == 3)` or `assert 3 == 3`

**Reply by spytheman** [reply]
The program is generated by the REPL. It has a heuristic, to determine what is an expression, and what is a statement, and it misfired for `assert 3==3`. I think that it  is a valid issue.
